config.go

```diff
@@ -576,10 +576,17 @@ func (s *Settings) GetGithubOauthStrings() ([]string, error) {
 	return tokens, nil
 }
 
+const (
+	Localhost = "localhost"
+)
+
 func (s *Settings) GetGithubOauthToken() (string, error) {
 	if s == nil {
 		return "", errors.New("not defined")
 	}
+	if s.DomainName == Localhost {
+		return Localhost, nil
```

Comment:
I'd prefer not to add code to prod that specifically cases on running locally. Some alternatives:
1) store a fake token in https://github.com/evergreen-ci/evergreen/blob/990f2a8d3f8a44d939aadea5ad9aae8d749e609d/testdata/local/admin.json#L5 so that we do have a token to return (even though it won't work), or 
2) Don't error if token doesn't exist when it's called for GetPatchedConfig, just log, and then only call GetCommitEvent if the token isn't empty.